cheatsheets/gleam-for-javascript-users.md

```diff
@@ -0,0 +1,700 @@
+  - [Case](#case)
+  - [Piping](#piping)
+
+
+## Comments
+
+### JavaScript
+
+In JavaScript, comments are written with a `//` prefix.
+
+```javascript
+// Hello, Joe!
+```
+
+Multi-line comments may be written like so:
+
+```javascript
+/*
+ * Hello, Joe!
+ */
+```
+
+In JavaScript, above `class` and `function` declarations there can be
+`docblocks` like so:
+
+```javascript
+/**
+ * A Bar class
+ */
+class Bar {}
+
+/**
+ * A quux function.
+ *
+ * @param {string} str    String passed to quux
+ * @returns {string}      An unprocessed string
+ */
+function quux(string) { return str; }
+```
+
+Documentation blocks (docblocks) are extracted into generated API
+documentation.
+
+### Gleam
+
+In Gleam comments are written with a `//` prefix.
+
+```gleam
+// Hello, Joe!
+```
+
+Comments starting with `///` are used to document the following statement.
+Comments starting with `////` are used to document the current module.
+
+```gleam
+//// This module is very important.
+
+/// The answer to life, the universe, and everything.
+const answer: Int = 42
+```
+
+## Variables
+
+In JavaScript, you can declare variables using `let`, `const`, or `var`. `let`
+and `var` variables can be reassigned, and `const`s cannot.
+
+#### JavaScript
+
+```javascript
+let size = 50;
+size = size + 100;
+
+const height = 60;
+height = height + 100; // Error!
+```
+
+#### Gleam
+
+Gleam has the `let` keyword before each variable assignment. Variables can't be
+mutated or reassigned. If a new variable has the same name as an existing
+variable, the new definition _shadows_ the old one, but the old variable is not
+overwritten:
+
+```gleam
+import gleam/io
+
+let size = 50
+let print_it = fn() { io.debug(size) }
+let size = size + 100
+let size = 1
+io.debug(size) // prints 1
+print_it() // prints 50
+```
+
+### Variables type annotations
+
+#### JavaScript/TypeScript
+
+In plain JavaScript there are no static types, but in TypeScript variables can
+optionally be annotated with types.
+
+```typescript
+const some_list: number[] = [1, 2, 3];
+```
+
+#### Gleam
+
+In Gleam type annotations can optionally be given when binding variables.
+
+```gleam
+let some_list: List(Int) = [1, 2, 3]
+```
+
+Gleam will check the type annotation to ensure that it matches the type of the
+assigned value. It does not need annotations to type-check your code, but you
+may find it useful to annotate variables to hint to the compiler that you want a
+specific type to be inferred.
+
+## Functions
+
+#### JavaScript
+
+In JavaScript, you can define functions with the `function` keyword, or assign
+anonymous functions to variables.
+
+```javascript
+function sum(x, y) {
+  return x + y;
+}
+
+const mul = (x, y) => x * y;
+```
+
+#### Gleam
+
+Gleam's functions are declared using a syntax similar to Rust. Gleam's anonymous
+functions have a similar syntax, just without the function name.
+
+```gleam
+pub fn sum(x, y) {
+  x + y
+}
+
+let mul = fn(x, y) { x * y }
+mul(1, 2)
```

Comment:
This code implies you can have top level expressions as if Gleam were a scripting language, but this is not the case

Reply:
Although its on the gleam tour already, I think it's worth repeating there is no **explicit return**, because many JS developers might not know other functional language.